Source/Core/VideoBackends/Software/Clipper.cpp

```diff
@@ -331,18 +331,34 @@ void ProcessTriangle(OutputVertexData* v0, OutputVertexData* v1, OutputVertexDat
   }
 }
 
-static void CopyVertex(OutputVertexData* dst, const OutputVertexData* src, float dx, float dy,
-                       unsigned int sOffset)
+constexpr std::array<float, 8> LINE_PT_TEX_OFFSETS = {
+    0, 1 / 16.f, 1 / 8.f, 1 / 4.f, 1 / 2.f, 1, 1, 1,
+};
+
+static void CopyLineVertex(OutputVertexData* dst, const OutputVertexData* src, float dx, float dy,
+                           u32 sOffset)
 {
+  dst->projectedPosition = src->projectedPosition;
   dst->screenPosition.x = src->screenPosition.x + dx;
   dst->screenPosition.y = src->screenPosition.y + dy;
   dst->screenPosition.z = src->screenPosition.z;
 
   dst->normal = src->normal;
   dst->color = src->color;
 
-  // todo - s offset
   dst->texCoords = src->texCoords;
+
+  if (LINE_PT_TEX_OFFSETS[sOffset] != 0)
```

Comment:
Is `sOffset` guaranteed to be in the range 0..7 here? Or do we want this to blow up if it isn't? (Or do we want to add a range check?)
Same for the other one in `CopyPointVertex`.

Reply:
Yes, it's guaranteed because it comes from a 3-bit field:

https://github.com/dolphin-emu/dolphin/blob/0029ca84b07f44e3ce8c80a44fa0285a966b34d3/Source/Core/VideoCommon/BPMemory.h#L577-L578

Thoughts on removing the parameter and directly using `bpmem.lineptwidth.pointoff`/`lineoff` in the function though?